homeassistant/components/evohome/water_heater.py

```diff
@@ -42,7 +44,9 @@ async def async_setup_platform(
     if discovery_info is None:
         return
 
-    broker = hass.data[DOMAIN]["broker"]
+    broker: EvoBroker = hass.data[DOMAIN]["broker"]
+
+    assert isinstance(broker.tcs.hotwater, evo.HotWater)  # mypy
```

Comment:
This is an upstream typing issue that should be fixed

Reply:
No. 

```
homeassistant/components/evohome/water_heater.py:51: error: Item "None" of "HotWater | None" has no attribute "TYPE"  [union-attr]
homeassistant/components/evohome/water_heater.py:52: error: Item "None" of "HotWater | None" has no attribute "dhwId"  [union-attr]
homeassistant/components/evohome/water_heater.py:55: error: Argument 2 to "EvoDHW" has incompatible type "HotWater | None"; expected "HotWater"  [arg-type]
```

For some systems, they don't have DHW - so type of tcs.dhw being `HotWater | None is correct`.

It can't be `None` is this block, because `async_setup_platform()` would not have been called in that case.

Hence the assert, which is a hint to HA, that can be safely ignored in the case of a `python -O`.